src/blocks/frontend/countdown/index.ts

```diff
@@ -105,42 +115,50 @@ class CountdownData {
 			}
 		);
 
-		this.onEndEvents = [ () => this.activateBehaviour(), () => this.activateActions() ];
+		// Add the default End events callbacks.
+		this.onEndEvents = [ () => this.activateBehavior(), () => this.activateActions() ];
 
+		/**
+		 * Modes
+		 * - Evergreen: acts like a timer that reset when it is finished.
+		 * - Interval: display only in the given interval.
+		 * - Default: display until the given date.
+		 */
 		switch ( this.mode ) {
 		case 'timer':
 
 			// Record when the user was last time on this page.
-			const lastVisitTimeRecord = localStorage.getItem( `${LAST_TIME_VISIT_ON_SITE_RECORD_SOURCE}-${this.elem.id}` );
+			const lastVisitTimeRecord = localStorage.getItem( `${LAST_TIME_VISIT_ON_SITE_RECORD_SOURCE}-${this.elem.id}` ) ?? '0';
 
-			// Record what was the timer value.
+			// Record with the timer value.
 			const timerValueRecorded = localStorage.getItem( `${TIMER_VALUE_FROM_LAST_TIME_VISIT_ON_SITE_RECORD_SOURCE}-${this.elem.id}` );
 
-			// Set the deadling based on the last visit.
+			// Record with the timezone
+			const timezoneRecorded = localStorage.getItem( `${TIMER_TIMEZONE_FROM_LAST_VISIT_ON_SITE}-${this.elem.id}` ) ?? '0';
+
+			// Set the deadline based on the last visit.
 			this.deadline = parseInt( lastVisitTimeRecord! ) + parseInt( this.timer );
 
-			// Check if the client is first time on the page.
+			/**
+			 * Reset conditions:
+			 * - the user is first time on the page
+			 * - the timer has reach the deadline
+			 * - the deadline time was changed
+			 * - the timezone has changed
+			 */
 			if (
 				! lastVisitTimeRecord ||
-				( ( parseInt( lastVisitTimeRecord ) + parseInt( this.timer ) - Date.now() ) > COUNTDOWN_RESET ) ||
-				timerValueRecorded !== this.timer
+				( 0 > ( parseInt( lastVisitTimeRecord ) + parseInt( this.timer ) - Date.now() ) ) ||
+				timerValueRecorded !== this.timer ||
+				timezoneRecorded != ( new Date() ).getTimezoneOffset().toString()
 			) {
```

Comment:
@HardeepAsrani, do you think we need more conditions beside those?

Reply:
Seems good enough to me.

Reply:
You can put it in QA if you're satisfied with the rest. And if you want we can test on the themeisle website to test.